service/efsPort.py

```diff
@@ -0,0 +1,630 @@
+    # Matches to item attribute "rigSize" on ship and rig items
+    SMALL = 1
+    MEDIUM = 2
+    LARGE = 3
+    CAPITAL = 4
+
+
+class EfsPort():
+    wepTestSet = {}
+
+    @staticmethod
+    def attrDirectMap(values, target, source):
+        for val in values:
+            target[val] = source.itemModifiedAttributes[val]
+
+    @staticmethod
+    def getT2MwdSpeed(fit, sFit):
+        fitID = fit.ID
+        propID = None
+        shipHasMedSlots = fit.ship.itemModifiedAttributes["medSlots"] > 0
+        shipPower = fit.ship.itemModifiedAttributes["powerOutput"]
+        # Monitors have a 99% reduction to prop mod power requirements
+        if fit.ship.name == "Monitor":
+            shipPower *= 100
+        rigSize = fit.ship.itemModifiedAttributes["rigSize"]
+        if not shipHasMedSlots:
+            return None
+
+        filterVal = Item.groupID == getGroup("Propulsion Module").ID
+        propMods = gamedata_session.query(Item).options().filter(filterVal).all()
+        mapPropData = lambda propName: \
+                      next(map(lambda propMod: {"id": propMod.typeID, "powerReq": propMod.attributes["power"].value},
+                               (filter(lambda mod: mod.name == propName, propMods))))
+        mwd5mn = mapPropData("5MN Microwarpdrive II")
+        mwd50mn = mapPropData("50MN Microwarpdrive II")
+        mwd500mn = mapPropData("500MN Microwarpdrive II")
+        mwd50000mn = mapPropData("50000MN Microwarpdrive II")
+        if rigSize == RigSize.SMALL or rigSize is None:
+            propID = mwd5mn["id"] if shipPower > mwd5mn["powerReq"] else None
+        elif rigSize == RigSize.MEDIUM:
+            propID = mwd50mn["id"] if shipPower > mwd50mn["powerReq"] else mwd5mn["id"]
+        elif rigSize == RigSize.LARGE:
+            propID = mwd500mn["id"] if shipPower > mwd500mn["powerReq"] else mwd50mn["id"]
+        elif rigSize == RigSize.CAPITAL:
+            propID = mwd50000mn["id"] if shipPower > mwd50000mn["powerReq"] else mwd500mn["id"]
+
+        if propID is None:
+            return None
+        sFit.appendModule(fitID, propID)
+        sFit.recalc(fit)
+        fit = eos.db.getFit(fitID)
+        mwdPropSpeed = fit.maxSpeed
+        mwdPosition = list(filter(lambda mod: mod.item and mod.item.ID == propID, fit.modules))[0].position
+        sFit.removeModule(fitID, mwdPosition)
+        sFit.recalc(fit)
+        fit = eos.db.getFit(fitID)
+        return mwdPropSpeed
+
+    @staticmethod
+    def getPropData(fit, sFit):
+        fitID = fit.ID
+        propGroupId = getGroup("Propulsion Module").ID
+        propMods = filter(lambda mod: mod.item and mod.item.groupID == propGroupId, fit.modules)
+        activePropWBloomFilter = lambda mod: mod.state > 0 and "signatureRadiusBonus" in mod.item.attributes
+        propWithBloom = next(filter(activePropWBloomFilter, propMods), None)
+        if propWithBloom is not None:
+            oldPropState = propWithBloom.state
+            propWithBloom.state = 0
```

Comment:
Probably best to import the `State` Enum and use that instead of the hardcoded value here, for explicitness

Reply:
Will do, I intended to use them where possible but I missed this one.